wp-content/plugins/core/src/Templates/Components/Deferred_Component.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare( strict_types=1 );
+
+namespace Tribe\Project\Templates\Components;
+
+class Deferred_Component extends \ArrayObject {
+	/**
+	 * @var string
+	 */
+	private $path;
+	/**
+	 * @var string|null
+	 */
+	private $name;
+
+	public function __construct( string $path, string $name = null, array $args = [] ) {
```

Comment:
This will also allow us to reduce quite a bit of component template noise with this container. +1